src/bin/signal_handler.rs

```diff
@@ -62,23 +69,48 @@ mod imp {
                         info!("{}", v)
                     }
                 }
-                SIGUSR2 => {
-                    if backup_path.is_empty() {
-                        info!("empty backup path, backup is disabled");
-                        continue;
-                    }
-
-                    info!("backup db to {}", backup_path);
-                    if let Err(e) = engine.backup_at(backup_path) {
-                        error!("fail to backup: {}", e);
-                    }
-                    info!("backup done");
-                }
+                SIGUSR2 => profile_memory(&profiling_memory),
                 // TODO: handle more signal
                 _ => unreachable!(),
             }
         }
     }
+
+    fn profile_memory(flag: &Arc<AtomicBool>) {
+        if flag.load(Ordering::SeqCst) {
+            warn!("last memory profiling has not finished yet.");
+            return;
+        }
+        unsafe {
+            if let Err(e) = jemallocator::mallctl_set(b"prof.dump\0", DUMP_FILE_NAME) {
```

Comment:
I don't think the allocation here can affect the profile result. 

`\0` is strange here. why use &[u8] but not &str for the arg?